packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2163,7 +2163,8 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
 
     final TextStyle hintStyle = _getInlineHintStyle(themeData, defaults);
     final String? hintText = decoration.hintText;
-    final Widget? hint = hintText == null ? null : AnimatedOpacity(
+    final bool adjustHeightForHintOnInput = !isEmpty && decoration.adjustHeightForHintOnInput;
+    final Widget? hint = (adjustHeightForHintOnInput || hintText == null) ? null : AnimatedOpacity(
```

Comment:
This change will break the fadeout animation. If the flag is set, then the hintText disappears immediately.

Reply:
@LongCatIsLooong Thank you for your review, I got it

Reply:
I think with the new implementation it will still disappear immediately without the fade out animation.

Reply:
@LongCatIsLooong Thank you for your review, I have updated~
![Simulator Screen Recording - iPhone 15 Pro Max - 2024-09-01 at 14 01 52](https://github.com/user-attachments/assets/cd947544-6b45-4ee2-869d-28def29cb27a)